src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -1,38 +1,143 @@
 #include "jit_read_tuples.hpp"
 
 #include "../jit_types.hpp"
+#include "all_type_variant.hpp"
 #include "expression/evaluation/expression_evaluator.hpp"
+#include "jit_expression.hpp"
 #include "resolve_type.hpp"
+#include "storage/segment_iterables/create_iterable_from_attribute_vector.hpp"
 #include "storage/segment_iterate.hpp"
 
 namespace opossum {
 
+namespace {
+
+struct CastedDictionary {
+  std::shared_ptr<const BaseDictionarySegment> dictionary_segment = nullptr;
+  std::shared_ptr<const PosList> pos_list = nullptr;
+};
+CastedDictionary get_dictionary_segment(const std::shared_ptr<const BaseSegment> segment) {
+  if (const auto dict_segment = std::dynamic_pointer_cast<const BaseDictionarySegment>(segment)) {
+    return {dict_segment};
+  }
+
+  const auto reference_segment = std::dynamic_pointer_cast<const ReferenceSegment>(segment);
+  if (!reference_segment) return {};
+
+  const auto pos_list = reference_segment->pos_list();
+  if (!pos_list->references_single_chunk() || pos_list->empty()) return {};
+
+  const auto referenced_chunk = reference_segment->referenced_table()->get_chunk(pos_list->common_chunk_id());
+  const auto referenced_segment = referenced_chunk->get_segment(reference_segment->referenced_column_id());
+  const auto dict_segment = std::dynamic_pointer_cast<const BaseDictionarySegment>(referenced_segment);
+  return {dict_segment, pos_list};
+}
+
+AllTypeVariant convert_variant_to_data_type(const AllTypeVariant& variant, const DataType requested_type) {
```

Comment:
If desired, this functionality could also be added to `all_type_variant.hpp `.

Reply:
How is this different from `type_cast`?